samza-azure/src/main/java/org/apache/samza/system/azureblob/BlobClientBuilderFactory.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.samza.system.azureblob;
+
+/**
+ * Constructs a new instance of {@link AzureBlobClientBuilder}
+ * from factory provided by configuration.
+ */
+public interface BlobClientBuilderFactory {
+  /**
+   * Create a new instance of {@link AzureBlobClientBuilder}
+   * @param systemName Name of the system
```

Comment:
Can you make the parameter description more intuitive to clarify their purpose and expected values.
Something like:
@param systemName The name of the system for which the blob client builder is being created.
@param azureUrlFormat The format of the Azure URL, which should conform to Azure's URL formatting requirements.
@param azureBlobConfig The configuration settings for Azure Blob, encapsulated in an {@link AzureBlobConfig} object. This includes details like account name, account key, and other relevant settings.